PWGGA/GammaConv/AliAnalysisTaskSigmaPlToProtonPiZero.cxx

```diff
@@ -513,165 +513,152 @@ void AliAnalysisTaskSigmaPlToProtonPiZero::UserExec(Option_t *)
-		nclus = fESD->GetNumberOfCaloClusters();
-		fHistNClusWoCuts[iCut]->Fill(nclus);
-		if(nclus == 0)  continue;
-		AliVCluster* clus = NULL;
-		for(Int_t i=0; i < nclus; ++i){
-			clus = new AliESDCaloCluster(*(AliESDCaloCluster*)fEvent->GetCaloCluster(i));
-			if(!clus) {
-				Printf("ERROR: Could not find clus %i",i);
-				continue;
-			}
-			fHistClusterEWOCuts[iCut]-> Fill(clus->E());
-			if((((AliCaloPhotonCuts*)fClusterCutArray->At(iCut))->ClusterIsSelected(clus,fESD,fMCEvent,fIsMC, 1.,i)) == kTRUE){
-				photon.push_back(clus);
-				fHistClusterE[iCut]-> Fill(clus->E());
-				nClusWCuts+= 1;
-			}
-			clus = NULL;
-		}
-		fHistNClusWCuts[iCut]->Fill(nClusWCuts);
-		//Paaring
-		TLorentzVector rekombinatedPi0;
-		TLorentzVector photonCandidate1;
-		TLorentzVector photonCandidate2;
-		TLorentzVector protonVektor;
-		TLorentzVector sigmaVektor;
-		Bool_t trueSigmaProton = kFALSE;
-		Bool_t trueSigmaPhoton1 = kFALSE;
-		Bool_t trueSigmaPhoton2 = kFALSE;
-		Bool_t mesonSelected = kFALSE;
-
-		if(proton.size() > 0 && photon.size() > 1){
-			for(unsigned int iProton = 0; iProton < proton.size(); ++iProton){
-				trueSigmaProton = kFALSE;
-				if(!proton[iProton]) {
-					Printf("ERROR: Could not find proton[%i][%i]",iCut,iProton);
+		if(proton.size() > 0){
+			//Find gammas
+			Int_t nclus                     = 0;
+			Int_t nClusWCuts                = 0;
+			nclus = fESD->GetNumberOfCaloClusters();
+			if(fHistNClusWoCuts[iCut]) fHistNClusWoCuts[iCut]->Fill(nclus);
+			if(nclus == 0)  continue;
+			AliVCluster* clus = NULL;
+			for(Int_t i=0; i < nclus; ++i){
+				clus = new AliESDCaloCluster(*(AliESDCaloCluster*)fEvent->GetCaloCluster(i));
+				if(!clus) {
+					Printf("ERROR: Could not find clus %i",i);
 					continue;
 				}
-				AliESDtrack* protonCandidate = proton[iProton];
-				protonVektor.SetPtEtaPhiM(protonCandidate->Pt(),protonCandidate->Eta(),protonCandidate->Phi(), 0.938272);
-				AliESDkink *kink=fESD->GetKink(TMath::Abs(protonCandidate->GetKinkIndex(0))-1);
-				if(!kink){
-					Printf("ERROR: Could not find kink[%i][%i]",iCut,iProton);
-					continue;
-				}
-				const TVector3 vposKink(kink->GetPosition());
-				Double_t trackKink[3] = {vposKink[0], vposKink[1], vposKink[2]};
-				Double_t kinkDistanceToVertex = sqrt(vposKink[0] * vposKink[0] + vposKink[1] * vposKink[1] + vposKink[2] * vposKink[2]);
-				if(fIsMC > 0){
-					trueSigmaProton = IsRealProtonKink(kink, fMCStack);
+				if(fHistClusterEWOCuts[iCut]) fHistClusterEWOCuts[iCut]-> Fill(clus->E());
+				if((((AliCaloPhotonCuts*)fClusterCutArray->At(iCut))->ClusterIsSelected(clus,fESD,fMCEvent,fIsMC, 1.,i)) == kTRUE){
+					photon.push_back(clus);
+					if(fHistClusterE[iCut]) fHistClusterE[iCut]-> Fill(clus->E());
+					nClusWCuts+= 1;
 				}
-				for(unsigned int iPhoton1 = 0; iPhoton1 < photon.size(); ++iPhoton1) {
-					trueSigmaPhoton1 = kFALSE;
-					if(!photon[iPhoton1]) {
-						Printf("ERROR: Could not find photon[%i][%i]",iCut,iPhoton1);
+				clus = NULL;
```

Comment:
What happens to the ESDCluster which is not used for the vector? Shouldn't it be deleted here?